WtWSMS/common/on_action/dlc/ep1/BP_court_language_on_actions.txt

```diff
@@ -46,3 +43,214 @@ bp_court_language_on_actions = {
 		}
 	}
 }
+
+on_title_gain = {
+	on_actions = {
+		BP_set_court_langauge
+	}
+}
+
+#Keep Gupta and Bureaucratic Court Languages
+BP_set_court_langauge = {
+	effect = {
+		if = {
+			limit = {
+				has_royal_court = yes
+				has_government = bureaucratic_government
+				scope:title.tier >= tier_kingdom
+				scope:title = root.primary_title #Fire only once per succession
+				primary_title = { has_variable = court_language }
+			}
+			set_court_language_from_effect = {
+				CULTURE = primary_title.var:court_language
+				CHARACTER = root
+			}
+		}
+		else_if = {
+			limit = {
+				has_royal_court = yes
+				scope:title.tier >= tier_kingdom
+				has_government = rajamandala_government
+				scope:title = root.primary_title #Fire only once per succession
+			}
+			set_court_language = language_sanskrit
+		}
+	}
+}
+
+on_game_start = {
+	on_actions = { setup_administrative_languages }
+}
+
+setup_administrative_languages = {
+	effect = {
+		#Set some court langauge variables for bureaucrats
+		title:e_wre = {
+			set_variable = {
+				name = court_language
+				value = culture:roman
+			}
+		}
+		title:e_byzantium = {
+			if = {
+				limit = { current_date < 610.10.5 }
+				set_variable = {
+					name = court_language
+					value = culture:roman
+				}
+			}
+			else = {
+				set_variable = {
+					name = court_language
+					value = culture:greek
+				}
+			}
+		}
+		title:k_italy = {
+			set_variable = {
+				name = court_language
+				value = culture:roman
+			}
+			holder = {
+				setup_the_court_language_effect = {
+					CHAR = this
+					LANGUAGE = language_latin
+				}
+			}
+		}
+		#Byzantines use Latin before Hearclius' reign
+		#Source: Davis, Leo Donald (1990). The First Seven Ecumenical Councils (325–787): Their History and Theology. Liturgical Press. p. 260. 
```

Comment:
This is not controversial, but in CK2 we had a decision about this, so I think the global flag should be included with the date, even if we don't port it for this update

Reply:
Depends on if we want to make use of CK3 court languages or not; seems a little odd to me to have Court Languages separate from Administrative Languages, but perhaps I'm alone in that.

Reply:
Yes, the decision should change the court language

Reply:
Which CK3 already has in Royal Court and which I integrated as part of that system.

Reply:
I think the decision about Greek administrative language should be accessible without having the DLC, since it wouldn't impact court language which doesn't exist without the DLC it shouldn't be an issue